deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/graph/vertex/impl/ElementWiseVertex.java

```diff
@@ -80,6 +84,12 @@ public INDArray doForward(boolean training) {
                     sum.addi(inputs[i]);
                 }
                 return sum;
+            case Average:
+                INDArray average = inputs[0].dup();
```

Comment:
Minor optimization: ```inputs[0].dup(inputs[0].ordering())``` to make it more likely to get good strides when doing the add ops...